octavia_proxy/tests/unit/api/drivers/elbv3/test_driver.py

```diff
@@ -81,6 +151,37 @@ def test_loadbalancers_qp(self):
             a='b'
         )
 
+    def test_loadbalancer_get(self):
+        self.driver.loadbalancer_get(self.sess, 'test', self.lb)
+        self.sess.vlb.find_load_balancer.assert_called_with(
+            name_or_id=self.lb, ignore_missing=True)
+
+    def test_loadbalancer_create(self):
+        self.driver.loadbalancer_create(self.sess, self.lb)
+        self.sess.vlb.create_load_balancer.assert_called_with(
+            **self.fakeCallCreate)
+
+    def test_loadbalancer_update(self):
+        attrs = {
+            'description': 'New Description',
+            'operating_status': 'ACTIVE',
+        }
+        self.driver.loadbalancer_update(self.sess, self.lb, attrs)
+        self.sess.vlb.update_load_balancer.assert_called_with(
+            self.lb.id, **attrs)
+
+    def test_loadbalancer_delete(self):
+        self.driver.loadbalancer_delete(self.sess, self.lb)
+        self.sess.vlb.delete_load_balancer.assert_called_with(self.lb.id)
+
+    def test_loadbalancer_delete_cascade(self):
+        self.driver.loadbalancer_delete(self.sess, self.lb, cascade=True)
+        self.sess.vlb.find_load_balancer.assert_called_with(
```

Comment:
and where is the difference to regular delete?

Reply:
in regular delete called only delete_load_balancer, in cascade find_load_balancer and then delete_load_balancer

Reply:
but the implementation is not like that. Ideally we should also test that delete for child resources are called

Reply:
how can i mock this child resources? i can't figure out

Reply:
https://github.com/opentelekomcloud-infra/octavia-proxy/blob/5576741149e78707801a4cdebc62fdce0d76a598/octavia_proxy/tests/unit/api/drivers/elbv3/test_driver.py#L120 - example of the mock setup. Means for the LB you need to enable mocks for querying and deleting sub resources. GET mocks need to return some fake data which you then verify on the DELETE mocks

Reply:
done